src/main/scala/edu/ie3/simona/io/grid/GridProvider.scala

```diff
@@ -29,18 +32,20 @@ object GridProvider {
       case GridSourceType.CSV =>
         gridDataSource.csvParams match {
           case Some(params) =>
-            CsvGridSource
-              .readGrid(
-                simulationName,
-                params.csvSep,
-                params.directoryPath,
-                new FileNamingStrategy()
-              )
-              .getOrElse(
-                throw new InitializationException(
-                  "Error while initializing CsvGridSource! Cannot proceed without a valid GridSource!"
+            val jointGridContainer = CsvJointGridContainerSource.read(
+              simulationName,
+              params.csvSep,
+              params.directoryPath
+            )
+
+            Try.apply(ValidationUtils.check(jointGridContainer)) match {
+              case Failure(exception) =>
+                logger.warn(
+                  s"Error while initializing CsvGridSource! Cannot proceed without a valid GridSource!",
+                  exception
```

Comment:
I think it suffices to log the exception message here, otherwise it looks like too much of a serious error if we let the whole stacktrace be printed

```suggestion
                  s"Error while initializing CsvGridSource! Cannot proceed without a valid GridSource! \n\t{}",
                  exception.getMessage
```